### ISSUE TITLE
Enforce limit of maximum 5 Collections per database

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/jsonapi#577]
Due to limits for maximum SAIs, we can only create up to 5 Collections per tenant/Database (note: not per namespace): this is to prevent failures on index creation during collection creation.

But currently we do not enforce this limit. We should.

**Comment by vkarpov15** [comment]
This restriction seems a bit strict. It won't break any of our sample apps, those only use 3-4 collections each, but this restriction would likely be a source of friction if you're looking to build a complete app using JSON API as the primary data store.

If you have multiple databases, can you work around this limitation?

**Comment by tatu-at-datastax** [comment]
Yes, multiple databases is the way to go. But the limit itself is due to technical limitations wrt indexing -- you can only have up to 50 SAIs per Database, and each Collection requires something like 9.
So it's not that tables are expensive, but indexes are.

Setting of 5 will be configurable using the usual system, and we'll change default if index limit is raised for Astra.

**Comment by tatu-at-datastax** [comment]
Initial PR only enforces 5-per-namespace, not database. Will re-file; rename this one
